lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -525,7 +547,7 @@ def __get_k8s_containers(self, ajs):
             # command line execution, separated by ;, which is what Galaxy does
             # to assemble the command.
             "command": [ajs.job_wrapper.shell],
-            "args": ["-c", ajs.job_file],
+            "args": ["-c", f"{ajs.job_file}; exit $(cat {ajs.exit_code_file})"],
```

Comment:
Before this, even if the tool failed, k8s saw the job as successful. Even if it was killed over OOM. This makes k8s aware that the tool failed (and makes the k8s Job fail accordingly). This then allows to retrieve, only for k8s failed jobs, additional status on the failure (OOM, walltime, etc).